nbextensions/bulkImportCell/bulkImportCell.js

```diff
@@ -452,6 +456,31 @@ define([
                 description: 'parent bus for BulkImportCell',
             });
             busEventManager.add(cellBus.on('delete-cell', () => deleteCell()));
+            controllerBus = runtime.bus().makeChannelBus({
+                description: 'An app cell widget',
+            });
+            controllerBus.on('update-param-state', (message) => {
+                updateParameterState(message.fileType, message.state);
+            });
+        }
+
+        /**
+         * Update the internal readiness state for the app cell parameters. When all fileTypes have
+         * a "complete" state, then they are all completely filled out and ready to run.
+         * @param {string} fileType - which filetype's app state to update
+         * @param {string} state - what the new ready state should be - one of complete, incomplete, error
+         */
+        function updateParameterState(fileType, state) {
+            model.setItem(['state', 'param', fileType], state);
+            let cellReady = true;
+            for (const state of Object.values(model.getItem('state.param'))) {
+                if (state !== 'complete') {
+                    cellReady = false;
+                    break;
+                }
+            }
+            if (cellReady) {
+            }
```

Comment:
Just a stub for now, will get filled out in the next PR. This is the part where the cell gets toggled into the "ready to run" state.